### PR TITLE
[DO NOT MERGE] Test PR for new benchmarking workflow

### PR DESCRIPTION
**Pull request by ianayl**
This PR exists so I can trigger and test workflows. I am currently working on a proof-of-concept workflow that benchmarks a PR's changes using sycl-bench.

Unfortunately, there is no real way to trigger a workflow and test without this PR, so thanks for your patience!

### CODE REVIEW

**Comment by sarnex** [comment]
@ianayl Please merge sycl/HEAD with your changes or we risk breaking CI for everyone, see recent internal email.

I'm cancelling the Windows jobs here for now, please don't rerun them until we have merged.

File: .github/workflows/sycl-benchmark.yml [hunk 1]
```diff
@@ -0,0 +1,215 @@
+name: Run benchmarks (using sycl-bench)
```
Comment (uditagarwal97) [inline .github/workflows/sycl-benchmark.yml:1]: I'd reword the workflow name. Something like `Run sycl-bench microbenchmarks`

File: .github/workflows/sycl-benchmark.yml [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+  workflow_dispatch:
+    inputs:
+      runner:
+        type: choice
+        options:
+          - '["Linux", "gen12"]'
+          - '["amdgpu"]'
+          - '["Linux", "arc"]'
+          - '["cts-cpu"]'
```
Comment (uditagarwal97) [inline .github/workflows/sycl-benchmark.yml:37]: You can remove the `[cts-cpu]` option.

File: .github/workflows/sycl-benchmark.yml [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    steps:
+    - name: Download SYCL toolchain
+      if: inputs.sycl_toolchain_artifact != '' && github.event_name != 'workflow_run'
+      uses: actions/download-artifact@v4
+      with:
+        name: ${{ inputs.sycl_toolchain_artifact }}
+    - name: Debug prints [workflow_run]
+      if: inputs.sycl_toolchain_artifact != '' && github.event_name == 'workflow_run'
+      run: |
+        pwd
```
Comment (uditagarwal97) [inline .github/workflows/sycl-benchmark.yml:86]: I think we can get rid of `pwd` and `ls`

File: .github/workflows/sycl-benchmark.yml (hunk 1 above)
Comment (ianayl) [reply .github/workflows/sycl-benchmark.yml:1]: Fixed in #14454

File: .github/workflows/sycl-benchmark.yml (hunk 3 above)
Comment (ianayl) [reply .github/workflows/sycl-benchmark.yml:86]: Fixed in #14454

**Comment by ianayl** [comment]
I have rewritten this test as a sub-option of sycl-linux-run-tests in #14454 , which should have fixes for Windows CI.
